database/merkledb/tree.go

```diff
@@ -0,0 +1,281 @@
+package merkledb
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database/versiondb"
+
+	"github.com/ava-labs/avalanchego/database/leveldb"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+// Tree holds the tree data
+type Tree struct {
+	closed      bool
+	persistence *Persistence
+}
+
+// Has returns whether the key exists in the tree
+func (t *Tree) Has(key []byte) (bool, error) {
+	if t.isClosed() != nil {
+		return false, t.isClosed()
+	}
+
+	rootNode, err := t.persistence.GetRootNode()
+	if err != nil {
+		return false, err
+	}
+
+	node, err := t.findNode(BytesToKey(key), rootNode)
+	if err != nil {
+		return false, err
+	}
```

Comment:
I'll change the name to traverseTree ? 
Will take advice on names :)
I'll add these comments to the function tho.

The `findNode` traverses the Tree and finds the most suited node for the traversal.
It traverses the tree trying to find the Node where that key might exist but doesn't guaraante that the key actualy exists, it's the callers responsability to handle that.

The reason for that is because it's used in all operations for tree traversal/finding where the node should exist.